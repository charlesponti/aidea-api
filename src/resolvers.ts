import Query from './queries';
import Mutation from './mutations';
import {Vote} from './data';

export default {
  Idea: {
    votes: (idea: any) => Vote.sum('vote', {where: {idea_id: idea.id}}),
  },
  Query,
  Mutation,
};
