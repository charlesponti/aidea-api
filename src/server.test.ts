import {ApolloServer} from 'apollo-server';
import typeDefs from './schema';
import resolvers from './resolvers';

const {createTestClient} = require('apollo-server-testing');

it('fetches single launch', async done => {
  // create a test server to test against, using our production typeDefs,
  // resolvers, and dataSources.
  const server = new ApolloServer({
    typeDefs,
    resolvers,
    context: () => ({user: {id: 1, email: 'a@a.a'}}),
  });

  // use the test server to create a query function
  const {query} = createTestClient(server);

  // run query against the server and snapshot the output
  const res = await query({query: '{ ideas { id title} }', variables: {id: 1}});
  expect(res.data).toMatchSnapshot();
  done();
});
