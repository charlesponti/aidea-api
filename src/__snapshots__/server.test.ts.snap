// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches single launch 1`] = `
Object {
  "ideas": Array [
    Object {
      "id": "1",
      "title": "foo",
    },
  ],
}
`;
